plugin/pkg/admission/podplacementrestriction/admission.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podplacementrestriction
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/controller/informers"
+	"k8s.io/kubernetes/pkg/util/tolerations"
+	"k8s.io/kubernetes/pkg/util/yaml"
+)
+
+func init() {
+	admission.RegisterPlugin("PodPlacementRestriction", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig := readConfig(config)
+		return NewPodTolerationsPlugin(client, pluginConfig.PodTolerationsPluginConfig), nil
+	})
+}
+
+var _ = admission.Interface(&podTolerationsPlugin{})
+var _ = admission.WantsInformerFactory(&podTolerationsPlugin{})
+
+type podTolerationsPlugin struct {
```

Comment:
this needs to be updated to use:
https://github.com/kubernetes/kubernetes/pull/39109

the actual configuration itself would need to be versioned similar to pr here:
https://github.com/kubernetes/kubernetes/pull/36765